index.js

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env node
+
+const core = require("@actions/core");
+const github = require("@actions/github");
+const { Octokit, App } = require("octokit");
+
+async function run() {
+    const githubToken = core.getInput("github_token");
+    const debug = core.getInput("debug") === "true" || false;
+
+    if (debug) {
+        console.log(`Event name: ${github.context.eventName}`);
+        console.log(
+            `Event payload: ${JSON.stringify(
+                github.context.payload,
+                undefined,
+                2
+            )}`
+        );
+    }
+
+    const octokit = new Octokit({ auth: githubToken });
+
+    if (github.context.eventName === "issue_comment") {
+        const body = github.context.payload.comment.body;
+        const authorAssociation =
+            github.context.payload.comment.author_association;
+        const pull_number = github.context.payload.issue.number;
```

Comment:
The `issue` field is used in some conditionals later on, so it might be good to destructure it for easier access.
```suggestion
        const { issue } = github.context.payload;
```

Reply:
part of next commit